pvlib/pvsystem.py

```diff
@@ -201,20 +201,21 @@ def __init__(self,
         if temperature_model_parameters is None:
             self.temperature_model_parameters = \
                 self._infer_temperature_model_params()
-            # TODO: in v0.8 check if an empty dict is returned and raise error
         else:
             self.temperature_model_parameters = temperature_model_parameters
 
-        # TODO: deprecated behavior if PVSystem.temperature_model_parameters
-        # are not specified. Remove in v0.8
-        if not any(self.temperature_model_parameters):
+        # warn user about change in default behavior in 0.9.
+        if all(x is None for x in
+               (temperature_model_parameters, module_type, racking_model)):
             warnings.warn(
-                'Required temperature_model_parameters is not specified '
-                'and parameters are not inferred from racking_model and '
-                'module_type. Reverting to deprecated default: SAPM cell '
-                'temperature model parameters for a glass/glass module in '
-                'open racking. In the future '
-                'PVSystem.temperature_model_parameters will be required',
+                'temperature_model_parameters, racking_model, and module_type '
+                'are not specified. Reverting to deprecated default: SAPM '
+                'cell temperature model parameters for a glass/glass module '
+                'in open racking. In v0.9 '
+                'PVSystem.temperature_model_parameters will be empty dict '
+                'unless set by temperature_model_parameters or valid '
+                'combination of racking_model and module_type. Pass non-None '
```

Comment:
```suggestion
                'in open racking. In v0.9, temperature_model_parameters or a valid '
                'combination of racking_model and module_type will be required. '
                ' Pass non-None '
```

I'd prefer to be more specific how to use v0.9.

Reply:
I'm ok with "will be required" if the warning is emitted from `PVSystem.sapm_celltemp`. I don't like it if the warning is emitted from the `PVSystem.__init__` because it suggests a positional argument.

Reply:
> `PVSystem()` has always been a valid, non-warned statement. There's never been a requirement that `PVSystem` is a complete description of a system according to some model. We can discuss changing that, but it's inconsistent to require temperature parameters and nothing else.

Agree, and point taken. I'm OK with the change proposed for `ModelChain.infer_temperature_model`.

Reply:
changed the text in the warning in the method